docs: improve doctests for complex number instances in `ndarray/base/unary`

aryan7071 opened the pull request:
---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes. report:
  - task: lint_filenames status: passed
  - task: lint_editorconfig status: passed
  - task: lint_markdown status: na
  - task: lint_package_json status: na
  - task: lint_repl_help status: na
  - task: lint_javascript_src status: passed
  - task: lint_javascript_cli status: na
  - task: lint_javascript_examples status: na
  - task: lint_javascript_tests status: na
  - task: lint_javascript_benchmarks status: na
  - task: lint_python status: na
  - task: lint_r status: na
  - task: lint_c_src status: na
  - task: lint_c_examples status: na
  - task: lint_c_benchmarks status: na
  - task: lint_c_tests_fixtures status: na
  - task: lint_shell status: na
  - task: lint_typescript_declarations status: passed
  - task: lint_typescript_tests status: na
  - task: lint_license_headers status: passed ---

Resolves part of #8641

## Description

This pull request updates documentation examples for `ndarray/base/unary/lib` to use complex number instance notation (e.g., `<Complex64>[ 10.0, 20.0 ]`) instead of decomposing values using `realf` and `imagf`, in accordance with the RFC guidance.

No functional code changes were made. Only documentation examples were updated.

## Related Issues

- #8641 

## Questions

No.

## Other

No.

## Checklist

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

### AI Assistance

-   [x] No
-   [ ] Yes

#### Disclosure

N/A

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

aryan7071 committed:
docs: improve doctests for complex number instances in ndarray/base/unary/lib

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

kgryte commented on lib/node_modules/@stdlib/ndarray/base/unary/lib/0d_accessors.js:
```diff
@@ -46,11 +46,9 @@
 *
 * function scale( z ) {
-*     return new Complex64( realf(z)*10.0, imagf(z)*10.0 );
+*     return new Complex64( z.re*10.0, z.im*10.0 );
```
@aryan7071 This change is not desired. We should still be using the functional accessors for retrieving real and imaginary components. That applies throughout this PR.

kgryte commented on lib/node_modules/@stdlib/ndarray/base/unary/lib/0d_accessors.js:
```diff
@@ -101,12 +99,7 @@
-*
-* var im = imagf( v );
-* // returns 40.0
+* // returns <Complex64>[ 30.0, 40.0 ]
```
This is the only type of change which is desired.

kgryte reviewed (changes requested):
In order for this PR to move forward, the `realf` and `imagf` imports need to be restored and the `scale` callback restored.

aryan7071 commented:
ok i am going to  revert this change

aryan7071 committed:
docs: improve doctests for complex number instances in ndarray/base/unary

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

kgryte commented on lib/node_modules/@stdlib/ndarray/base/unary/lib/3d_blocked_accessors.js:
```diff
@@ -53,11 +53,9 @@ var blockSize = require( '@stdlib/ndarray/base/unary-tiling-block-size' );
 * var Complex64 = require( '@stdlib/complex/float32/ctor' );
-* var realf = require( '@stdlib/complex/float32/real' );
-* var imagf = require( '@stdlib/complex/float32/imag' );
 *
```
```suggestion
* var realf = require( '@stdlib/complex/float32/real' );
* var imagf = require( '@stdlib/complex/float32/imag' );
*
```

kgryte commented on lib/node_modules/@stdlib/ndarray/base/unary/lib/3d_blocked_accessors.js:
```diff
@@ -53,11 +53,9 @@ var blockSize = require( '@stdlib/ndarray/base/unary-tiling-block-size' );
 *
 * function scale( z ) {
-*     return new Complex64( realf(z)*10.0, imagf(z)*10.0 );
+*     return new Complex64( z.re*10.0, z.im*10.0 );
```
```suggestion
*     return new Complex64( realf(z)*10.0, imagf(z)*10.0 );
```

kgryte committed:
Apply suggestions from code review

Signed-off-by: Athan <kgryte@gmail.com>

kgryte reviewed (approved):
LGTM
